lib/cocoapods/generator/xcconfig/pod_xcconfig.rb

```diff
@@ -72,10 +72,10 @@ def generate
           recursive_dependent_targets = target.recursive_dependent_targets
           @xcconfig.merge! XCConfigHelper.settings_for_dependent_targets(target, recursive_dependent_targets, @test_xcconfig)
           if @test_xcconfig
-            test_dependent_targets = [target, *target.test_dependent_targets]
+            test_dependent_targets = [target, *target.recursive_test_dependent_targets].uniq
```

Comment:
This does not invoke `all_test_dependent_targets` because on line 73 above we've already added recursive dependent targets.